cocotb/drivers/__init__.py

```diff
@@ -252,45 +249,44 @@ def __init__(self, entity: SimHandleBase, name: Optional[str], clock: SimHandleB
         # Give this instance a unique name
         self.name = name if index is None else "%s_%d" % (name, index)
 
-    @coroutine
-    def _driver_send(self, transaction, sync: bool = True) -> None:
+    async def _driver_send(self, transaction, sync: bool = True) -> None:
         """Implementation for BusDriver.
 
         Args:
             transaction: The transaction to send.
             sync: Synchronize the transfer by waiting for a rising edge.
         """
         if sync:
-            yield RisingEdge(self.clock)
+            await RisingEdge(self.clock)
         self.bus <= transaction
 
     @coroutine
-    def _wait_for_signal(self, signal):
+    async def _wait_for_signal(self, signal):
```

Comment:
I think this is "protected", and called by downstream possibly-third-party drivers